### ISSUE TITLE
[REVIEW]: Launcher: A simple tool for executing high throughput computing workloads

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#289]
**Submitting author:** @lwilson (<a href="http://orcid.org/0000-0002-0686-3152">Lucas Wilson</a>)
**Repository:** <a href="https://github.com/TACC/launcher" target ="_blank">https://github.com/TACC/launcher</a>
**Version:** v3.1
**Editor:** @danielskatz
**Reviewer:** @kc9qey
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.841337" target="_blank">10.5281/zenodo.841337</a>

## Status

[![status](http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7/status.svg)](http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7"><img src="http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7/status.svg)](http://joss.theoj.org/papers/7b5df63cd8a40f557d66051695d300a7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/TACC/launcher" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/TACC/launcher">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v3.1)?
- [x] **Authorship:** Has the submitting author (@lwilson) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @kc9qey it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by danielskatz** [comment]
@kc9qey - this is ready for you to review. You should review the reviewer instructions on http://joss.theoj.org/about. Let me know if you have any problems.

**Comment by kc9qey** [comment]
TACC/launcher Functionality documentation needed.
    #21 opened a minute ago by kc9qey
    TACC/launcher Along with example usage, missing automated test(s).
    #20 opened 2 minutes ago by kc9qey
    TACC/launcher missing example usage
    #19 opened 3 minutes ago by kc9qey
    TACC/launcher pre-requisite needs mentioned
    #18 opened 5 minutes ago by kc9qey

**Comment by danielskatz** [comment]
just adding links to make it easier to follow the previous comment:

> TACC/launcher Functionality documentation needed.
> https://github.com/TACC/launcher/issues/21 opened a minute ago by kc9qey
> TACC/launcher Along with example usage, missing automated test(s).
> https://github.com/TACC/launcher/issues/20 opened 2 minutes ago by kc9qey
> TACC/launcher missing example usage
> https://github.com/TACC/launcher/issues/19 opened 3 minutes ago by kc9qey
> TACC/launcher pre-requisite needs mentioned
> https://github.com/TACC/launcher/issues/18 opened 5 minutes ago by kc9qey

**Comment by danielskatz** [comment]
Hi @lwilson - It looks like you've resolves one of the four issues above, but still need to work on the other three.  I'm just checking on the status of them, so it's clear who has the ball currently.

**Comment by danielskatz** [comment]
Hi @lwilson - just another ping...

**Comment by danielskatz** [comment]
@lwilson - the review process is waiting on your resolution of the last 3 issues created by @kc9qey

**Comment by danielskatz** [comment]
Hi @lwilson - this is another ping...

**Comment by danielskatz** [comment]
I have pinged @lwilson offline

**Comment by danielskatz** [comment]
@arfon - it's been over a month since we've heard from the author @lwilson here. Any thoughts?

**Comment by lwilson** [comment]
Dan and Galen,

I’m still alive! Sorry for the delay on this – I’ll be pushing changes and updating those tickets this week.

________________________________
Lucas A. Wilson, Ph.D.
Director, Training and Professional Development
Texas Advanced Computing Center
The University of Texas at Austin
ACB 2.278
+1 512 232 7351
lwilson@tacc.utexas.edu<mailto:lwilson@tacc.utexas.edu>

**Comment by danielskatz** [comment]
great!

**Comment by danielskatz** [comment]
@lwilson - not sure what "this" week meant... :)

**Comment by lwilson** [comment]
It's apparently a "baker's" week (it has 1 extra day)!

**Comment by lwilson** [comment]
We have made some updates on:
https://github.com/TACC/launcher/issues/21
https://github.com/TACC/launcher/issues/20
https://github.com/TACC/launcher/issues/19

**Comment by danielskatz** [comment]
@kc9qey - back to you...

**Comment by danielskatz** [comment]
@kc9qey - just another ping ...

**Comment by kc9qey** [comment]
all good.

**Comment by kc9qey** [comment]
I don't have anything else to review on this.  They hit all my points I
think.  I was in the hospital last week getting an important upgrade.

**Comment by danielskatz** [comment]
@kc9qey - there are still a couple of boxes in the checklist that are not checked.

**Comment by kc9qey** [comment]
I didn't see any automated tests, so left unchecked.
Performance claims for me are a "won't do" --also left unchecked.

**Comment by danielskatz** [comment]
@kc9qey

If there are no performance claims, you can check that box.

Are you also saying that there are no automated tests or manual steps described so that the function of the software can be verified?  If so, then this box cannot be checked, and we need to go back to @lwilson

**Comment by kc9qey** [comment]
"manual steps described so that the function of the software can be verified" , i can take that for the automated tests.

**Comment by danielskatz** [comment]
@kc9qey 0- thanks for the review work!

**Comment by danielskatz** [comment]
@arfon - I now hand this over to you for final steps

**Comment by danielskatz** [comment]
@arfon - ping...

**Comment by arfon** [comment]
@lwilson - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by lwilson** [comment]
10.5281/zenodo.841337

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.841337 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.841337" target="_blank">10.5281/zenodo.841337</a> is the archive.

**Comment by arfon** [comment]
@kc9qey many thanks for your review and thanks to @danielskatz for editing this submission ✨ 

@lwilson - your paper is now accepted into JOSS. Your DOI is http://dx.doi.org/10.21105/joss.00289 ⚡️ 🚀 💥
